tests/integration/pure_cairo_vm/contracts/roll_warp_tester.cairo

```diff
@@ -3,49 +3,71 @@
+    let (block_timestamp) = stored_block_timestamp.read();
     return (res=block_timestamp);
 }
 
 @view
-func block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (block_number) = get_block_number();
+func stored_block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_number) = stored_block_number.read();
     return (res=block_number);
 }
 
-@storage_var
-func stored_block_timestamp() -> (res: felt) {
+@view
+func timestamp_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_timestamp) = get_block_timestamp();
+    return (res=block_timestamp);
 }
 
-@storage_var
-func stored_block_number() -> (res: felt) {
+@view
+func block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_number) = get_block_number();
+    return (res=block_number);
 }
 
-@view
-func block_timestamp_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+@external
+func block_timestamp_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
     let (block_timestamp) = get_block_timestamp();
     stored_block_timestamp.write(block_timestamp);
 
-    return (res=block_timestamp);
+    return ();
 }
 
-@view
-func block_number_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+@external
+func block_number_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
     let (block_number) = get_block_number();
     stored_block_number.write(block_number);
-    return (res=block_number);
+
+    return ();
 }
 
-@view
-func stored_block_timestamp_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (block_timestamp) = stored_block_timestamp.read();
-    return (res=block_timestamp);
+@contract_interface
+namespace TimestampAndBlockNumberTesterContract {
+    func block_timestamp_setter() {
+    }
+
+    func block_number_setter() {
+    }
 }
 
-@view
-func stored_block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (block_number) = stored_block_number.read();
-    return (res=block_number);
-}
\ No newline at end of file
+@external
+func call_block_timestamp_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(contract_address: felt) {
+    TimestampAndBlockNumberTesterContract.block_timestamp_setter(contract_address=contract_address);
+    return ();
+}
+
+@external
+func call_block_number_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(contract_address: felt) {
+    TimestampAndBlockNumberTesterContract.block_number_setter(contract_address=contract_address);
+    return ();
+}
```

Comment:
Maybe that should be in a separate file?

Reply:
Do you mean the functions calling external contract? I don't see a reason to do that honestly. Especially since we already want to split the contract into 2.